src/vm/amd64/asmhelpers.S

```diff
@@ -0,0 +1,188 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+.intel_syntax noprefix
+#include "unixasmmacros.inc"
+#include "asmconstants.h"
+
+#define real4 dword
+#define real8 qword
+
+//
+//    file: profile.cpp
+//    typedef struct _PROFILE_PLATFORM_SPECIFIC_DATA
```

Comment:
We also need to store the six integer argument register in this struct. As I have mentioned before, this is not needed on Windows since there are always four slots on stack reserved for storing argument registers right before the return address, but this is not the case on Unix. So the profiler needs to have some place to get those registers from.